pandas/tests/indexes/test_interval.py

```diff
@@ -327,171 +335,298 @@ def test_get_item(self):
+        assert index.slice_locs(end=Interval(0, 2)) == (0, 3)
+        assert index.slice_locs(
+            start=Interval(2, 4), end=Interval(0, 2)) == (0, 3)
+
+        # sorted duplicates
+        index = IntervalIndex.from_tuples([(0, 2), (0, 2), (2, 4)])
+
+        assert index.slice_locs(
+            start=Interval(0, 2), end=Interval(2, 4)) == (0, 3)
+        assert index.slice_locs(start=Interval(0, 2)) == (0, 3)
+        assert index.slice_locs(end=Interval(2, 4)) == (0, 3)
+        assert index.slice_locs(end=Interval(0, 2)) == (0, 2)
+        assert index.slice_locs(
+            start=Interval(2, 4), end=Interval(0, 2)) == (2, 2)
+
+        # unsorted duplicates
+        index = IntervalIndex.from_tuples([(0, 2), (2, 4), (0, 2)])
+
+        pytest.raises(KeyError, index.slice_locs(
+            start=Interval(0, 2), end=Interval(2, 4)))
+        pytest.raises(KeyError, index.slice_locs(start=Interval(0, 2)))
+        assert index.slice_locs(end=Interval(2, 4)) == (0, 2)
+        pytest.raises(KeyError, index.slice_locs(end=Interval(0, 2)))
+        pytest.raises(KeyError, index.slice_locs(
+            start=Interval(2, 4), end=Interval(0, 2)))
+
+        # another unsorted duplicates
+        index = IntervalIndex.from_tuples([(0, 2), (0, 2), (2, 4), (1, 3)])
+
+        assert index.slice_locs(
+            start=Interval(0, 2), end=Interval(2, 4)) == (0, 3)
+        assert index.slice_locs(start=Interval(0, 2)) == (0, 4)
+        assert index.slice_locs(end=Interval(2, 4)) == (0, 3)
+        assert index.slice_locs(end=Interval(0, 2)) == (0, 2)
+        assert index.slice_locs(
+            start=Interval(2, 4), end=Interval(0, 2)) == (2, 2)
+
+    @pytest.mark.xfail(reason="new indexing tests for issue 16316")
+    def test_slice_locs_with_ints_and_floats_updated_behavior(self):
+
+        queries = [[0, 1], [0, 2], [0, 3], [3, 1], [3, 4], [0, 4]]
+
+        # increasing non-overlapping
+        index = IntervalIndex.from_tuples([(0, 1), (1, 2), (3, 4)])
 
-        assert not i.contains(20)
-        assert not i.contains(-20)
+        assert index.slice_locs(0, 1) == (0, 1)
+        assert index.slice_locs(0, 2) == (0, 2)
+        assert index.slice_locs(0, 3) == (0, 2)
+        assert index.slice_locs(3, 1) == (2, 1)
+        assert index.slice_locs(3, 4) == (2, 3)
+        assert index.slice_locs(0, 4) == (0, 3)
+
+        # decreasing non-overlapping
+        index = IntervalIndex.from_tuples([(3, 4), (1, 2), (0, 1)])
+        assert index.slice_locs(0, 1) == (3, 2)
+        assert index.slice_locs(0, 2) == (3, 1)
+        assert index.slice_locs(0, 3) == (3, 1)
+        assert index.slice_locs(3, 1) == (1, 2)
+        assert index.slice_locs(3, 4) == (1, 0)
+        assert index.slice_locs(0, 4) == (3, 0)
+
+        # increasing overlapping
+        index = IntervalIndex.from_tuples([(0, 2), (1, 3), (2, 4)])
+        for query in queries:
+            pytest.raises(KeyError, index.slice_locs, query)
```

Comment:
Is there a reason you changed this back from InvalidIndexError to KeyError ?

Reply:
There's a helpful KeyError message in pandas right now, specific to this issue. No need to change it I'd say.